coinaddrng/validation.py

```diff
@@ -726,3 +727,88 @@ def prefixtodec(prefix):
         total += prefix[-i]*multiplier
         multiplier *= 256
     return total+prefix[-1]
+
+
+@attr.s(frozen=True, slots=True, cmp=False)
+@implementer(IValidator)
+class SS58Validator(ValidatorBase):
+
+    name = 'SS58Check'
+    valid_ss58_format = None
+
+    def validate(self):
+        try:
+            self._ss58_decode(self.request.address, valid_ss58_format=self.valid_ss58_format)
+        except ValueError:
+            return False
+
+        return True
+
+    # https://github.com/paritytech/substrate/wiki/External-Address-Format-(SS58)
+    def _ss58_decode(self, address: bytes, valid_ss58_format: Optional[int] = None) -> str:
+        address_decoded = base58check.b58decode(address)
+
+        if address_decoded[0] & 0b0100_0000:
```

Comment:
Vytahl bych tohle do separatni staticke metody, neco jako `_decode_address_format(address: bytes) -> SS58AddressFormat`.

`SS58AddressFormat` by byla separatni `attr.s` trida:

```
@attr.s(frozen=True, slots=True, auto_attribs=true)
class SS58AddressFormat:
    format_length: int
    format: int
```

+ Jsou to dobra jmena tech promennych?
+ Adresy zacinaji `0b1...` jsou rezervovane pro budouci pouziti, chceme je mit ted podporovat? Nebo vyhodime `UnsupportedError`? @galvanizze